packages/aws-cdk-lib/aws-lambda-nodejs/test/bundling.test.ts

```diff
@@ -245,6 +245,7 @@ test('esbuild bundling with esbuild options', () => {
       '--resolve-extensions': '.ts,.js',
       '--splitting': true,
       '--keep-names': '',
+      '--out-extension': '.js=.mjs',
```

Comment:
why do we need this?

Reply:
I have added this to check if the code is working fine. `out-extension` is one of the fields that uses `:`

Reply:
so why didn't we add the others here?

Reply:
There is a list of esbuildArgs that uses ':' to assign the flags. Here is the [list](https://github.com/aws/aws-cdk/pull/29167/files#diff-d3a78437ce44ac758a33052c1ce663976f465681199c2c85cc42b8b57832b08eR421) with all of them. The reason I chose `--out-extension` to test the code is because it was the one among the list that could be set without modifying much of the test code or without breaking any existing functionality.